### PR TITLE
Use custom router to remove external dependencies

### PR DESCRIPTION
**Pull request by spuun**
### WHAT is this pull request doing?
We used an [external http router](https://github.com/tbrand/router.cr). This PR adds a custom router and removes the external dependency.

### HOW can this pull request be tested?
Run specs.

### CODE REVIEW

**Comment by carlhoerberg** [comment]
Fantastic! 

missed removed a require though:
```
#0 2.929 In src/lavinmq/http/http_server.cr:3:1
#0 2.929 
#0 2.929  3 | require "router"
#0 2.929      ^
#0 2.929 Error: can't find file 'router'
```

File: src/lavinmq/http/router.cr [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    getter method, fragments, action
+
+    def initialize(@method : String, @path : String, @action : Action)
+      @fragments = Fragments.create(@path)
+    end
+  end
+
+  @_routes = Routes.new
+
+  {% for method in %w[delete get options patch post put] %}
```
Comment (carlhoerberg) [inline src/lavinmq/http/router.cr:102]: i guess we don't use `HEAD` but for completeness: 
```suggestion
  {% for method in %w[delete get options patch post put head] %}
```

Comment (spuun) [reply src/lavinmq/http/router.cr]: Actually we do (for static content), but we don't test it (it seems)!

**Comment by carlhoerberg** [comment]
Doesn't compile iwth `-Dwith_bake` or in `--release` mode: 

```
#39 15.90 In src/lavinmq/http/controller/static.cr:23:14
#39 15.90 
#39 15.90  23 | {{ run("./static/bake", PUBLIC_DIR) }}
#39 15.90          ^--
#39 15.90 Error: Error executing run (exit code: 1): ./static/bake 'Path["/usr/src/lavinmq_1.2.14-18-g726bb610/src/lavinmq/http/controller/../../../../static"].expand'
#39 15.90 
#39 15.90 
#39 15.90 stderr:
#39 15.90 
#39 15.90     Unhandled exception: Error opening directory: 'Path[\"/usr/src/lavinmq_1.2.14-18-g726bb610/src/lavinmq/http/controller/../../../../static\"].expand': No such file or directory (File::NotFoundError)
```

**Comment by spuun** [comment]
> Doesn't compile iwth `-Dwith_bake` or in `--release` mode:
> 
> ```
> #39 15.90 In src/lavinmq/http/controller/static.cr:23:14
> #39 15.90 
> #39 15.90  23 | {{ run("./static/bake", PUBLIC_DIR) }}
> #39 15.90          ^--
> #39 15.90 Error: Error executing run (exit code: 1): ./static/bake 'Path["/usr/src/lavinmq_1.2.14-18-g726bb610/src/lavinmq/http/controller/../../../../static"].expand'
> #39 15.90 
> #39 15.90 
> #39 15.90 stderr:
> #39 15.90 
> #39 15.90     Unhandled exception: Error opening directory: 'Path[\"/usr/src/lavinmq_1.2.14-18-g726bb610/src/lavinmq/http/controller/../../../../static\"].expand': No such file or directory (File::NotFoundError)
> ```

Right! Will fix. That `Path` I added was to get absolute paths in errors.

**Comment by spuun** [comment]
Note to self: run `shards prune` after you remove dependencies!

**Review by carlhoerberg — APPROVED** [review]
Very cool!

**Pull request merged**